proto-gen/api_v2/codec.go

```diff
@@ -0,0 +1,45 @@
+package api_v2
+
+import (
+	"github.com/gogo/protobuf/proto"
+	"google.golang.org/grpc/encoding"
+	gproto "google.golang.org/protobuf/proto"
+	"reflect"
+	"strings"
+)
+
+func init() {
+	encoding.RegisterCodec(newCodec())
+}
+
+// gogoCodec forces the use of gogo proto marshalling/unmarshalling for
+// proto over grpc
+type gogoCodec struct {
+}
+
+func newCodec() *gogoCodec {
+	return &gogoCodec{}
+}
+
+// Name implements encoding.Codec
+func (c *gogoCodec) Name() string {
+	return "proto"
+}
+
+// Marshal implements encoding.Codec
+func (c *gogoCodec) Marshal(v interface{}) ([]byte, error) {
+	t := reflect.TypeOf(v)
+	if strings.Contains(t.String(), "Empty") {
```

Comment:
This does the trick to pass the grpc storage test. This is obviously not good. if we go forward we should whitelist jaeger types to use gogo and use standard impl by default. 

cc) @joe-elliott

Reply:
Yeah, that's pretty rough. I understand why that works, but we're really holding things together with shoestrings and duct tape here. At the least it needs a comment to explain why it exists.

The good news is that Jaeger tests are robust enough we can feel confident in these changes, but I think someone is going to have to take on a major project to get us off of gogoproto for Jaeger's long term health.

Reply:
The idea behind this is to use gogo marshaller only for Jaeger types.

Reply:
Added comment and condition to use gogo only for jaeger types - based on the import path.